conans/client/generators/cmake_common.py

```diff
@@ -246,6 +246,8 @@ def generate_targets_section(dependencies):
         #     /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
         set(CMAKE_SKIP_RPATH 1)  # AVOID RPATH FOR *.dylib, ALL LIBS BETWEEN THEM AND THE EXE
                                  # SHOULD BE ON THE LINKER RESOLVER PATH (./ IS ONE OF THEM)
+        # Policy CMP0068
+        set(CMAKE_SKIP_INSTALL_RPATH 1) # In previous versions to 3.9 it is explicitly necessary
```

Comment:
According to the CMake [doc](https://cmake.org/cmake/help/v3.9/policy/CMP0068.html) it's 3.9 and newer.

Reply:
So is it wrong? If I understood it correctly, that is what the PR is doing, setting it (for cmake < 3.9), so it doesn't RPATH on install (it was a previous bug)

Reply:
Sorry I was only talking about the comment...

Reply:
The comment is mentioning versions prior to    3.9, but it should mention 3.9 and newer versions

Reply:
I still think the comment is ok, in previous versions, 3.7, 3.8, it is necessary to add that set explicitely, otherwise RPATHS are not skipped. In 3.9 and newers, this variable is simply not used, not necessary. I guess some language issue, not native speaker here :)

Reply:
I am not either so I may be wrong too :)

But the doc of `CMAKE_SKIP_RPATH` is clear about one thing:

>To omit RPATH in the install step, but not the build step, use CMAKE_SKIP_INSTALL_RPATH instead.

This implies that the added lines are not useful, `CMAKE_SKIP_RPATH` affect both installed and non-installed binaries.

The change brought by CMP0068 is that both `CMAKE_SKIP_RPATH` and `CMAKE_SKIP_INSTALL_RPATH` does **no longer affect** the `install_name` of a target.

Citing the doc:

>If SKIP_BUILD_RPATH, CMAKE_SKIP_RPATH or CMAKE_SKIP_INSTALL_RPATH were used to strip the directory portion of the install_name of a target, one may set INSTALL_NAME_DIR="" instead.

So I'd suggest replacing the merged lines with:

```cmake
set(CMAKE_INSTALL_NAME_DIR "")
```

PS:

One thing that I found curious is the Conan option `SKIP_RPATH`. When set to `1`, it does the opposite of what CMake does.

Could this option be renamed to something less misleading? It would break some people, but there is some days left before 1.0 :D

Reply:
I think we need the feedback from @lasote too here :)

Reply:
@theodelrieu is right about both things. I didn't understand right the new policy.

I suggest to remove the introduced line `CMAKE_SKIP_INSTALL_RPATH` and adjust the policy to the old behavior `cmake_policy(SET CMP0068 OLD)`.

About the conan parameter `SKIP_RPATH`, it tries to say `SKIP_THE_SKIP_RPATH` LoL. I agree that maybe we should change it, we could maintain the old name and introduce a new parameter. The default behavior should be the same, Conan skipping the rpaths, I think it the most wanted approach. So, suggestions are welcome, `KEEP_RPATHS`?

Reply:
> I suggest to remove the introduced line CMAKE_SKIP_INSTALL_RPATH and adjust the policy to the old behavior cmake_policy(SET CMP0068 OLD).

I've never played with `cmake_policy`, but it might be better/clearer than setting `INSTALL_NAME_DIR`. Your choice :)

>About the conan parameter SKIP_RPATH, it tries to say SKIP_THE_SKIP_RPATH LoL.

Yes that's what I thought too, it made me laugh :D. `KEEP_RPATHS` looks good to me!